internal/controller/issues.go

```diff
@@ -0,0 +1,301 @@
+		}
+		return fetchErr
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("exceeded retries fetching issues: %w", err)
+	}
+
+	r.Log.Info("Fetched issues successfully")
+	return allIssues, nil
+}
+
+// CloseIssue closes the issue on GitHub Repo.
+func (r *GithubIssueReconciler) CloseIssue(ctx context.Context, owner, repo string, platformIssue *git.Issue) error {
+	if platformIssue == nil {
+		return fmt.Errorf("cannot close issue: issue is nil")
+	}
+
+	// Close the issue using the generic IssueClient
+	closedIssue, err := r.IssueClient.Close(ctx, owner, repo, platformIssue.Number)
+	if err != nil {
+		return fmt.Errorf("failed to close issue: %v", err)
+	}
+
+	// Log the URL of the closed issue
+	r.Log.Info(fmt.Sprintf("Closed issue: %s", closedIssue.URL))
+	return nil
+}
+
+// CreateIssue creates a new issue in the repository
+func (r *GithubIssueReconciler) CreateIssue(ctx context.Context, owner, repo string, issueObject *issuesv1alpha1.GithubIssue) error {
+	// Create a new issue using the generic IssueClient
+	createdIssue, err := r.IssueClient.Create(ctx, owner, repo, issueObject.Spec.Title, issueObject.Spec.Description)
+	if err != nil {
+		return fmt.Errorf("failed to create issue: %v", err)
+	}
+
+	// Log the URL of the created issue
+	r.Log.Info(fmt.Sprintf("Created issue: %s", createdIssue.URL))
+	return nil
+}
+
+// EditIssue edits the description of an existing issue in the repository
+func (r *GithubIssueReconciler) EditIssue(ctx context.Context, owner, repo string, issueObject *issuesv1alpha1.GithubIssue, issueNumber int) error {
+	// Edit the issue using the generic IssueClient
+	editedIssue, err := r.IssueClient.Edit(ctx, owner, repo, issueNumber, issueObject.Spec.Description)
+	if err != nil {
+		return fmt.Errorf("failed to edit issue: %v", err)
+	}
+
+	// Log the URL of the edited issue
+	r.Log.Info(fmt.Sprintf("Edited issue: %s", editedIssue.URL))
+	return nil
+}
+
+// FindIssue finds a specific issue in the repository by title
+func (r *GithubIssueReconciler) FindIssue(ctx context.Context, owner, repo string, issue *issuesv1alpha1.GithubIssue) (*git.Issue, error) {
+	// Fetch all issues using the generic fetchAllIssues function
+	allIssues, err := r.fetchAllIssues(ctx, owner, repo)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching issues: %v", err)
+	}
+
+	// Extract the title from the GithubIssue object and search for it
+	return searchForIssue(issue.Spec.Title, allIssues), nil
+}
+
+// ParseRepoURL parses a repository URL and extracts the owner and repository name.
+// Returns an error if the URL format is invalid.
+func ParseRepoURL(repoURL string) (string, string, error) {
+	parts := strings.Split(repoURL, "/")
+	if len(parts) < 5 {
+		return "", "", fmt.Errorf("invalid repository URL: %s", repoURL)
+	}
+	return parts[3], parts[4], nil
+}
+
+func (r *GithubIssueReconciler) handleNewIssue(ctx context.Context, owner, repo string, issueObject *issuesv1alpha1.GithubIssue) (ctrl.Result, error) {
```

Comment:
Missing godoc above function. Remove inline redundant comments here and everywhere else